File: code/generate_strategies/IR(summary)/java/java/function_networkx_minimum_spanning_tree.java
package com.example;

import org.jgrapht.Graph;
import org.jgrapht.alg.spanning.KruskalMinimumSpanningTree;
import org.jgrapht.graph.DefaultWeightedEdge;
import org.jgrapht.graph.SimpleWeightedGraph;

public class NetworkXExample {

    
    public float compute_minimum_spanning_tree(String u, String v, float weight, String x, String y, Float weight2) {
        
        Graph<String, DefaultWeightedEdge> graph = new SimpleWeightedGraph<>(DefaultWeightedEdge.class);

        {
            
            graph.addVertex(u);
            graph.addVertex(v);
            
            DefaultWeightedEdge e1 = graph.addEdge(u, v);
            graph.setEdgeWeight(e1, weight);

            
            if (x != null && !x.isEmpty() && y != null && !y.isEmpty() && weight2 != null && weight2 != 0.0f) {
                graph.addVertex(x);
                graph.addVertex(y);
                DefaultWeightedEdge e2 = graph.addEdge(x, y);
                graph.setEdgeWeight(e2, weight2);
            }

            
            
            KruskalMinimumSpanningTree<String, DefaultWeightedEdge> mstAlgo = new KruskalMinimumSpanningTree<>(graph);
            double totalWeight = mstAlgo.getSpanningTree().getWeight();
            return (float) totalWeight;
        }
    }
}

File: code/generate_strategies/IR(summary)/cpp/cpp/function_networkx_minimum_spanning_tree.cpp

#include <boost/graph/adjacency_list.hpp>
#include <boost/graph/kruskal_min_spanning_tree.hpp>
#include <string>

class NetworkXExample {
public:
    template<typename T1, typename T2, typename T3, typename T4>
    float compute_minimum_spanning_tree(const T1& u, const T2& v, float weight,
                                      const T3& x = T3(), const T4& y = T4(), 
                                      float weight2 = 0) {
        using EdgeProperty = boost::property<boost::edge_weight_t, float>;
        using Graph = boost::adjacency_list<boost::vecS, boost::vecS, 
                                          boost::undirectedS,
                                          boost::no_property,
                                          EdgeProperty>;
        Graph g;

        
        add_edge(0, 1, weight, g);

        
        if (!std::string(x).empty() && !std::string(y).empty() && weight2 != 0) {
            add_edge(1, 2, weight2, g);
        }

        
        std::vector<boost::graph_traits<Graph>::edge_descriptor> mst;
        kruskal_minimum_spanning_tree(g, std::back_inserter(mst));

        
        float total_weight = 0;
        boost::property_map<Graph, boost::edge_weight_t>::type weight_map 
            = get(boost::edge_weight, g);
        
        for (const auto& edge : mst) {
            total_weight += boost::get(weight_map, edge);
        }

        return total_weight;
    }
};